core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -84,6 +84,8 @@ class BrokerServer(
   val supportedFeatures: util.Map[String, VersionRange]
 ) extends KafkaBroker {
 
+  override def brokerState: BrokerState = currentState()
```

Comment:
This is a bit broken. If kraft doesn't use the `_brokerState` field, we should not have it.

Reply:
Yeah, the same thought occurred to me, but I wanted to get a PR out there ASAP.  Will fix.